sale_loyalty_order_line_link/models/sale_order.py

```diff
@@ -7,36 +7,16 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    def _get_reward_values_product(self, program):
-        """Add the link to the program in the discount line"""
-        res = super()._get_reward_values_product(program)
-        res["coupon_program_id"] = program.id
-        return res
-
-    def _get_reward_values_discount(self, program):
-        """Add the link to the program in the discount line. The original method returns
-        a dict.values() to support multiple taxes. We can safely return a list instead
-        as the main method just iterates over the result of these auxiliar products to
-        write vals. https://git.io/J88As
-        """
-        res = list(super()._get_reward_values_discount(program))
-        [r.update(coupon_program_id=program.id) for r in res]
-        return res
-
     def write(self, vals):
         """We're looking for reward lines set in the write, from those, we'll get
         the proper ids to write them into the related lines"""
         res = super().write(vals)
-        reward_lines = [
-            x[2]
-            for x in vals.get("order_line", [])
-            if len(x) > 2 and x[0] == 0 and x[2].get("is_reward_line")
-        ]
+        reward_lines = self.order_line.filtered(
+            lambda x: x.loyalty_program_id and x.is_reward_line
+        )
```

Comment:
Not using the ORM is deliberate here for performance reasons. This change isn't needed.

**Anyway**, I'm thinking that now that we have the base module setting the reward for us maybe we could simplify the logic triggering the _link_reward_lines as a compute that depends on `reward_id`.

Reply:
'is_reward_line' doesn't come as a key in the sale.order write as it did in v15, so I made the change.

Reply:
Then let's convert these fields into computed store and trigger them from `reward_id`

Reply:
You mean convert loyalty_program_id and is_reward_line to computed?

Reply:
No, the links between lines